spec.bs

```diff
@@ -651,6 +677,169 @@ traversables</a> section.
+
+<div algorithm=focusing-steps-patch>
+  Modify the [=focusing steps=] to take a new optional [=boolean=] argument <dfn
+  lt="focus-unfenced">unfenced</dfn> that defaults to false.
+
+  Add a new step after step 3 of the algorithm (that changes new focus target) that reads:
+
+  4. If |new focus target| is a [=fenced navigable container=] with non-null
+     [=fenced navigable container/fenced navigable=], then set <var>new focus target</var> to the
+     [=fenced navigable container/fenced navigable=]'s [=navigable/active document=].
+
+  Add a new step after step 8 (step 7 in the existing version) of the algorithm (that defines |old
+  chain| and |new chain|) that reads:
+
+  9. If [=focus-unfenced|unfenced=] is false, |new chain| [=list/contains=] an element |element|
+     whose [=node navigable=]'s [=navigable/traversable navigable=] is a [=fenced navigable
+     container/fenced navigable=], and |old chain| does not also [=list/contain=] |element|, then
+     return.
+
+     Note: This is how we bail-out early just before calling the [=focus update steps=], in the case
+     where focus is trying to cross the fence.
+</div>
+
+<h4 id=focusing-steps-callers>Changes to the callers of the [=focusing steps=] algorithm</h4>
+
+Modify the user agent sentence after the algorithm steps to read:
+
+User agents must [=immediately=] run the [=focusing steps=] for a [=focusable area=] or
+[=navigable=] |candidate| with [=focus-unfenced|unfenced=] set to true whenever the user attempts to
+move the focus to |candidate|.
+
+<div algorithm=access-key-patch>
+  Modify the action of the [=accesskey attribute command=] algorithm to be:
+
+  1. Run the [=focusing steps=] for the element with [=focus-unfenced|unfenced=] set to true.
+
+  1. <a>Fire a `click` event</a> at the element.
+</div>
+
+<div algorithm=activation-patch>
+  Modify the behavior when a user [=activation|activates=] a [=click focusable=] [=focusable area=]
+  to be:
+
+  When a user [=activation|activates=] a [=click focusable=] [=focusable area=], the user agent must
+  run the [=focusing steps=] on the [=focusable area=] with <var ignore=''>focus trigger</var> set
+  to "`click`" and [=focus-unfenced|unfenced=] set to true.
+</div>
+
+<div algorithm=sequential-focus-patch>
+  Modify step 6 of the [=sequential focus navigation=] algorithm to be:
+
+  6. If <var>candidate</var> is not null, then run the [=focusing steps=] for
+    <var>candidate</var> with [=focus-unfenced|unfenced=] set to true and return.
+</div>
+
+<div algorithm=hide-popover-patch>
+  Modify step 10 of the [=hide popover algorithm=] to read:
+
+  <ol>
+    <li value="10">
+      <p>If <var>previouslyFocusedElement</var> is not null, then:</p>
+
+      <ol>
+      <li><p>Set <var ignore=''>element</var>'s [=previously focused element=] to null.</p></li>
+
+      <li><p>If <var ignore=''>focusPreviousElement</var> is true, then run the [=focusing steps=] for
```

Comment:
This step is identical to the one in the spec, and so is the above. Does this patch actually do anything?

Reply:
I forgot to add the unfenced part to this, good catch.